react-app/client/src/ExcelFunctionality/ExportToCSV.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { Button } from '@mui/material';
+import * as FileSaver from "file-saver";
+import * as XLSX from "xlsx";
+
+type Props = {
+  csvData: Object[];
+  fileName: string;
+  chooseRows: Object[];
+};
+
+export const ExportCSV: React.FC<Props> = ({ csvData, fileName, chooseRows }) => {
+  const fileType =
+    "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;charset=UTF-8";
+  const fileExtension = ".xlsx";
+
+  const newData = [];
+  if (chooseRows.length > 0) {
+    for(let i = 0; i < chooseRows.length; i++) {
+      newData.push(csvData[Number(chooseRows[i]) - 1]);
+    }
+    csvData = newData;
+  }
+
+  const exportToCSV = (csvData: Object[], fileName: string) => {
+  console.log(csvData)
```

Comment:
Nit: Indentation is off for this block of code.

Reply:
Fixed in recent pull request